src/poetry/core/masonry/builders/builder.py

```diff
@@ -70,12 +65,7 @@ def __init__(
         for include in self._package.include:
             formats = include.get("format", [])
 
-            if (
-                formats
-                and self.format
-                and self.format not in formats
-                and not ignore_packages_formats
-            ):
+            if formats and self.format and self.format not in formats:
```

Comment:
just simplification: no caller ever sets this parameter true.

Reply:
Let's not remove it for that reason. Core is to be treated as a library for now.

Reply:
Are you sure that's the right call?  Of course this can go back; but accruing dead code doesn't seem desirable to me?

Reply:
of course I intended to remove the parameter on the `__init__()` method as well as the implementation, leaving one but not the other makes no sense.

I'm going to push a commit that completes the removal of this parameter, and then ask you to confirm whether you really want it putting back.  

IMO it's better to clean these things up, but I'm happy to defer if maintainer opinion is definitely otherwise.

Reply:
In an ideal world, everything has test coverage, we are working towards better test coverage - however some of the older code paths do not have coverage. This is normal for a project like this. If we go by "remove everything untested", then a lot of poetry must go ;)  And to be fair, simply because core itself or poetry does not set this value today by no means imply that this is not used by a third-party code depending on poetry-core. In particular builders are something that is expected to be extended on.

For now, i think the call to keep the logic as is, is the least worst option. Bear in mind that this PR's scope is to introduce type checking, not alter logic.

I suspect the team would be happy to entertain targetted proposals removing "dead" code. But I would appreciate if we did that in a separate change set so we can evaluate the impact rather than shoehorning it into a type fix change.

I appreciate your opinion on this, and personally I can't really think of a reason why this flag would be valuable - but I can imagine others having a different view. As maintainers, we need to balance these. But better discussed separate to this.

Reply:
I put it back, this isn't important enough to argue about!  Thanks for the discussion, hopefully this is now good to merge.